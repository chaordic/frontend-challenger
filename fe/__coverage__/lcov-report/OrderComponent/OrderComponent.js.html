<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for OrderComponent/OrderComponent.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">OrderComponent</a> OrderComponent.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Fragment } from 'react';
import { string, array, func } from 'prop-types';
import cssModules from 'react-css-modules';
import ShipDetails from '../ShipDetails/ShipDetails';
&nbsp;
import styles from './styles.scss';
&nbsp;
const OrderComponent = ({ pointOfSale, fulfillments, createdAt, id, onToggle }) =&gt; (
  &lt;Fragment&gt;
    &lt;section className={styles.main}&gt;
      &lt;h1&gt;Dados do pedido&lt;/h1&gt;
      &lt;article&gt;
        {createdAt !== undefined &amp;&amp;
          &lt;section&gt;
            &lt;p className={styles.addressTitle}&gt;Comprado em&lt;/p&gt;
            &lt;p className={styles.address}&gt;{createdAt}&lt;/p&gt;
          &lt;/section&gt;
        }
&nbsp;
        {pointOfSale !== undefined &amp;&amp;
          &lt;section&gt;
            &lt;p className={styles.addressTitle}&gt;Ponto de venda&lt;/p&gt;
            &lt;p className={styles.address}&gt;{pointOfSale}&lt;/p&gt;
          &lt;/section&gt;
        }
      &lt;/article&gt;
    &lt;/section&gt;
    {fulfillments !== undefined &amp;&amp; Object.values(fulfillments).map(<span class="fstat-no" title="function not covered" >f </span>=&gt; (
<span class="cstat-no" title="statement not covered" >      &lt;ShipDetails</span>
        id={id}
        fulfillment={f}
        onToggle={onToggle}
      /&gt;
    ))}
  &lt;/Fragment&gt;
);
&nbsp;
OrderComponent.propTypes = {
  id: string,
  pointOfSale: string,
  createdAt: string,
  fulfillments: array,
  onToggle: func,
};
&nbsp;
OrderComponent.defaultProps = {
  id: '',
  createdAt: '',
  pointOfSale: '',
  fulfillments: [],
  onToggle: <span class="fstat-no" title="function not covered" >()</span> =&gt; {},
};
&nbsp;
export default cssModules(OrderComponent, styles, { allowMultiple: true });
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 11 2019 23:02:12 GMT+0000 (WET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
