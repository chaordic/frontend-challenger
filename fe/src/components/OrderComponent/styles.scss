@import '../../global-styles/settings.global.scss';

.main {
    -webkit-box-shadow: var(--shadow);
    -moz-box-shadow: var(--shadow);
    box-shadow: var(--shadow);
    padding: 20px;
    margin: 10px 20px;
}

.main > article {
    display: grid;
    grid-column-gap: 1rem;
    grid-template-columns: 3fr 3fr 3fr 3fr;
}

.main > h1 {
    padding: 0 0 10px 0;
    font-size: 18px;
    font-weight: 500;
}
