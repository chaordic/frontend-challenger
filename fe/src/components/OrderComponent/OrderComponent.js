import React, { Fragment } from 'react';
import { string, array, func } from 'prop-types';
import cssModules from 'react-css-modules';
import ShipDetails from '../ShipDetails/ShipDetails';

import styles from './styles.scss';

const OrderComponent = ({ pointOfSale, fulfillments, createdAt, id, onToggle }) => (
  <Fragment>
    <section className={styles.main}>
      <h1>Dados do pedido</h1>
      <article>
        {createdAt !== undefined &&
          <section>
            <p className={styles.addressTitle}>Comprado em</p>
            <p className={styles.address}>{createdAt}</p>
          </section>
        }

        {pointOfSale !== undefined &&
          <section>
            <p className={styles.addressTitle}>Ponto de venda</p>
            <p className={styles.address}>{pointOfSale}</p>
          </section>
        }
      </article>
    </section>
    {fulfillments !== undefined && Object.values(fulfillments).map(f => (
      <ShipDetails
        id={id}
        fulfillment={f}
        onToggle={onToggle}
      />
    ))}
  </Fragment>
);

OrderComponent.propTypes = {
  id: string,
  pointOfSale: string,
  createdAt: string,
  fulfillments: array,
  onToggle: func,
};

OrderComponent.defaultProps = {
  id: '',
  createdAt: '',
  pointOfSale: '',
  fulfillments: [],
  onToggle: () => {},
};

export default cssModules(OrderComponent, styles, { allowMultiple: true });
