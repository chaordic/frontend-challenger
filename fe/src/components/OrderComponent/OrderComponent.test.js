import React from 'react';
import renderer from 'react-test-renderer';
import OrderComponent from './OrderComponent';

const props = {
  formatAddress: jest.fn(),
  availableButton: true,
};

describe('(Component) OrderComponent', () => {
  it('should render an OrderComponent with result', () => {
    const tree = renderer.create(<OrderComponent {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
