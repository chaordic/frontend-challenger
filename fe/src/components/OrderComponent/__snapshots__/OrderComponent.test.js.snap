// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OrderComponent should render an OrderComponent with result 1`] = `
<section>
  <h1>
    Dados do pedido
  </h1>
  <article>
    <section>
      <p>
        Comprado em
      </p>
      <p>
        
      </p>
    </section>
    <section>
      <p>
        Ponto de venda
      </p>
      <p>
        
      </p>
    </section>
  </article>
</section>
`;
